### PR TITLE
Assignment/03

### PR DESCRIPTION
**Pull request by anandchavan24**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,43 +1,21 @@
...
-  };
 
   return (
     <div className="app">
-    <h1>Todo List</h1>
-    {loading && <p>Loading...</p>}
-    {error && <div className="error-banner">{error}</div>}
-    {todos && <TodoList todos={todos} markCompleted={markCompleted} deleteTodo={deleteTodo} />}
-  </div>
+      <BrowserRouter>
```
Comment (Rushikesh-josh) [inline src/App.tsx:11]: Please check difference between Browserrouter and Memoryrouter

File: src/Components/Common/addFetch.tsx
Comment (Rushikesh-josh) [inline src/Components/Common/addFetch.tsx:1]: what is the use of this file if it's returning nothing

File: src/Components/Common/useFetchById.tsx
Comment (Rushikesh-josh) [inline src/Components/Common/useFetchById.tsx:1]: please check this changes is already part of assignment no 2

File: src/Components/Feature/AddToDo.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import React, { useState } from 'react';
+
+const AddTodo = () => {
+  const [title, setTitle] = useState('');
+  const [body, setBody] = useState('');
```
Comment (Rushikesh-josh) [inline src/Components/Feature/AddToDo.tsx:7]: check is there any alternative way to take form data without storing it in the state

File: src/Components/Feature/ToDoList.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import { useEffect, useState } from 'react';
+import { IToDoResp } from '../../Shared/types'
+import Todo from './ToDo'
+import useFetch from '../Common/useFetch';
+import { Link } from 'react-router-dom';
```
Comment (Rushikesh-josh) [inline src/Components/Feature/ToDoList.tsx]: What is the use of Link in this component? If you are not using please remove it

File: src/Components/Feature/TodoItemDetail.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import useFetchById from "../Common/useFetchById";
+import Todo from "./ToDo";
+import { useParams } from "react-router-dom";
+
+const TodoItemDetail = () =>{
+    const { id } = useParams();
+
+    const [todos, setTodos] = useState<IToDoResp | null>(null);
+
+    const { data, loading, error }:{ data:IToDoResp | null; loading: boolean; error: string | null } = useFetchById('https://jsonplaceholder.typicode.com/todos/'+id);
```
Comment (Rushikesh-josh) [inline src/Components/Feature/TodoItemDetail.tsx]: define the type for { data:IToDoResp | null; loading: boolean; error: string | null }
and use it here

File: src/Components/Feature/TodoItemDetail.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  
+    useEffect(() => {
+      if (data) {
+        setTodos(data);
+      }
+    }, [data]);
+  
+    const markCompleted = (id: number) => {
```
Comment (Rushikesh-josh) [inline src/Components/Feature/TodoItemDetail.tsx]: all the handlers is repeated in this file please create reusable handlers in helper file

File: src/App.tsx (hunk 1 above)
Comment (anandchavan24) [reply src/App.tsx:11]: **BrowserRouter** for web applications where you need browser navigation features, and use **MemoryRouter** for non-browser environments or isolated routing needs within your React application.

File: src/Components/Common/addFetch.tsx
Comment (anandchavan24) [reply src/Components/Common/addFetch.tsx:1]: I have added resuable code

File: src/Components/Common/useFetchById.tsx
Comment (anandchavan24) [reply src/Components/Common/useFetchById.tsx:1]: No 

We need to fetch data by Id

File: src/Components/Feature/TodoItemDetail.tsx (hunk 5 above)
Comment (anandchavan24) [reply src/Components/Feature/TodoItemDetail.tsx]: I have removed where its do not needed

File: src/Components/Feature/ToDoList.tsx (hunk 3 above)
Comment (anandchavan24) [reply src/Components/Feature/ToDoList.tsx]: In React, Link is typically used with React Router to create navigation links within a single-page application (SPA). It's part of React Router's API and allows you to create declarative navigation without causing a full page refresh.

**Review by Rushikesh-josh — APPROVED** [review]

**Pull request merged**